packages/react-kit/src/components/Header/Header.tsx

```diff
@@ -0,0 +1,145 @@
+import { IconButton } from '@components/Button';
+import clsx from 'clsx';
+import FocusTrap from 'focus-trap-react';
+import React, { useEffect, useState } from 'react';
+import { CloseSVG, HamburgerMenuSVG } from './HeaderIcons';
+import { NavHeaderProps } from './props';
+
+/**
+ * @see {@link https://live-storybook-react-kit.appa.pantheon.site/?path=/docs/header-header--docs}
+ */
+export const Header = ({
+	Logo,
+	mainNavItems,
+	secondaryNavItems,
+	overlayStyles,
+	navbarStyles,
+	focusTrapOptions,
+}: NavHeaderProps) => {
+	const [isOpen, setIsOpen] = useState(false);
+	const [windowWidth, setWindowWidth] = useState(window.innerWidth);
+	const handleOpen = () => setIsOpen((prev) => !prev);
+
+	const MAX_MOBILE_NAV_WIDTH = 1023;
+
+	useEffect(() => {
+		const handleKeyUp = (e: KeyboardEvent) => {
+			if (e.key === 'Escape' || e.key === 'Esc') {
+				setIsOpen(false);
+			}
+		};
+		const handleResize = () => {
+			setWindowWidth(window.innerWidth);
+			windowWidth > MAX_MOBILE_NAV_WIDTH && setIsOpen(false);
+		};
+		setWindowWidth(window.innerWidth);
+
+		window.addEventListener('resize', handleResize);
+		window.addEventListener('keyup', handleKeyUp);
+
+		return () => {
+			window.removeEventListener('resize', handleResize);
+			window.removeEventListener('keyup', handleKeyUp);
+		};
+	}, [windowWidth]);
+
+	/**
+	 * Need to use a ref here so the FocusTrap can work properly.
+	 */
+	const NavList = React.forwardRef<
+		React.ElementRef<'nav'>,
+		Pick<NavHeaderProps, 'mainNavItems' | 'secondaryNavItems'>
+	>(({ mainNavItems, secondaryNavItems }, ref) => {
+		const NAVLIST_STYLES = {
+			nav: clsx('rk-flex rk-items-center'),
+			ul: clsx(
+				'lg:rk-menu-horizontal',
+				isOpen && windowWidth <= MAX_MOBILE_NAV_WIDTH
+					? 'rk-menu-vertical rk-w-full'
+					: 'rk-hidden rk-flex-wrap lg:rk-flex',
+			),
+		};
+
+		return (
+			<header
+				ref={ref}
+				className={clsx(
+					navbarStyles,
+					'rk-z-10 rk-flex rk-w-full rk-items-center rk-py-4',
+				)}
+			>
+				<Logo />
+				<div
+					className={clsx(
+						overlayStyles,
+						'rk-w-full rk-justify-around',
+						isOpen && windowWidth <= MAX_MOBILE_NAV_WIDTH
+							? 'rk-absolute rk-bottom-0 rk-left-0 rk-z-10 rk-h-[calc(100%_-_72px)] rk-flex-col rk-overflow-y-auto rk-overflow-x-hidden rk-px-4 rk-pb-4 rk-pt-8 sm:rk-px-6 lg:rk-px-12'
+							: 'rk-hidden rk-flex-wrap lg:rk-flex',
+					)}
+				>
+					<nav
+						className={clsx(NAVLIST_STYLES.nav, 'rk-mr-auto lg:rk-w-2/3')}
```

Comment:
Big improvement here, but 2/3 / 1/3 seems to give a little too much space between the nav elements.

<img width="1761" alt="flex-between" src="https://github.com/pantheon-systems/decoupled-kit-js/assets/889478/6e27ac1d-9c78-452a-87cc-f207123f1be7">

Could we get away with something like 3/4 / 1/4?